core/src/test/scala/org/broadinstitute/dsde/rawls/monitor/workspace/runners/LandingZoneCreationStatusRunnerSpec.scala

```diff
@@ -0,0 +1,455 @@
+    when(
+      billingRepository.updateCreationStatus(
+        ArgumentMatchers.eq(billingProjectName),
+        ArgumentMatchers.eq(CreationStatuses.Error),
+        ArgumentMatchers.any[Some[String]]()
+      )
+    ).thenAnswer { invocation =>
+      val message: Option[String] = invocation.getArgument(2)
+      assert(message.get.contains(billingProjectName.value))
+      assert(message.get.toLowerCase.contains("no user email"))
+      Future.successful(1)
+    }
+    val runner = new LandingZoneCreationStatusRunner(
+      mock[SamDAO],
+      mock[WorkspaceManagerDAO],
+      billingRepository,
+      mock[GoogleServicesDAO]
+    )
+    whenReady(runner.run(monitorRecord))(_ shouldBe WorkspaceManagerResourceMonitorRecord.Complete)
+
+    verify(billingRepository).updateCreationStatus(
+      ArgumentMatchers.eq(billingProjectName),
+      ArgumentMatchers.eq(CreationStatuses.Error),
+      ArgumentMatchers.any[Some[String]]()
+    )
+
+  }
+
+  it should "set an error in the billing project and return job as incomplete if the user context cannot be created" in {
+    val userEmail = "user@email.com"
+    val billingProjectName = RawlsBillingProjectName(UUID.randomUUID().toString)
+    val monitorRecord = WorkspaceManagerResourceMonitorRecord(
+      UUID.randomUUID(),
+      WorkspaceManagerResourceMonitorRecord.JobType.AzureLandingZoneResult,
+      None,
+      Some(billingProjectName.value),
+      Some(userEmail),
+      new Timestamp(Instant.now().toEpochMilli)
+    )
+    val billingRepository = mock[BillingRepository]
+    when(
+      billingRepository.updateCreationStatus(
+        ArgumentMatchers.eq(billingProjectName),
+        ArgumentMatchers.eq(CreationStatuses.Error),
+        ArgumentMatchers.any[Some[String]]()
+      )
+    ).thenAnswer { invocation =>
+      val message: Option[String] = invocation.getArgument(2)
+      assert(message.get.toLowerCase.contains("request context"))
+      assert(message.get.contains(userEmail))
+      Future.successful(1)
+    }
+    val runner =
+      spy(
+        new LandingZoneCreationStatusRunner(mock[SamDAO],
+                                            mock[WorkspaceManagerDAO],
+                                            billingRepository,
+                                            mock[GoogleServicesDAO]
+        )
+      )
+    doReturn(Future.failed(new org.broadinstitute.dsde.workbench.client.sam.ApiException()))
+      .when(runner)
+      .getUserCtx(ArgumentMatchers.eq(userEmail))(ArgumentMatchers.any())
+
+    whenReady(runner.run(monitorRecord))(_ shouldBe WorkspaceManagerResourceMonitorRecord.Incomplete)
+    verify(billingRepository).updateCreationStatus(
+      ArgumentMatchers.eq(billingProjectName),
+      ArgumentMatchers.eq(CreationStatuses.Error),
+      ArgumentMatchers.any[Some[String]]()
+    )
+  }
+  // TODO: test throwing a random exception from somewhere weird where it wouldn't be handled, like the failure message builder,
```

Comment:
Is this something you still want to do?

Reply:
No: the structure changed so that it's irrelevant - I'll remove the comment